plugins/module_utils/job.py

```diff
@@ -76,9 +73,52 @@ def _get_job_output(job_id="*", owner="*", job_name="*", dd_name=""):
                 str(rc), str(err)
             )
         )
-    if not out:
-        raise RuntimeError("Failed to retrieve job output. No job output found.")
-    jobs = _parse_jobs(out)
+    jobs = []
+    if out:
+        jobs = _parse_jobs(out)
+    if not jobs:
+        jobs = _job_not_found(job_id, owner, job_name, dd_name)
+
+    return jobs
+
+
+def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
+    jobs = []
+
+    job = {}
+
+    job["job_id"] = job_id
+    job["job_name"] = job_name
+    job["subsystem"] = None
+    job["system"] = None
+    job["owner"] = None
+
+    job["ret_code"] = {}
+    job["ret_code"]["msg"] = "Job Not Found"
+    job["ret_code"]["code"] = None
+    job["ret_code"]["msg_code"] = "00"
+    job["ret_code"]["msg_txt"] = "The job could not be found"
+
+    job["class"] = ""
+    job["content_type"] = ""
+
+    job["ddnames"] = []
+    dd = {}
+    dd["ddname"] = dd_name
+    dd["record_count"] = "0"
+    dd["id"] = ""
+    dd["stepname"] = "NOTFOUND"
+    dd["procstep"] = ""
+    dd["byte_count"] = "0"
+    job["ddnames"].append(dd)
+
+    if ovrr is not None:
```

Comment:
This error case (the plural one) seems to be missing the "job["ret_code"]["code"] = None" property, maybe I am misunderstanding why it was left off or maybe it is a coding error.

Reply:
the plural case comes after the singular case, which already set the ret_code/code to None